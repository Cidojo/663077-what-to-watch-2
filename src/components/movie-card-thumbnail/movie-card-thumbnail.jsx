import * as React from 'react';
import * as PropTypes from 'prop-types';

const MovieCardThumbnail = (props) => {
  const {card, onThumbnailClick, onThumbnailMouseOver} = props;

  return (
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOver={onThumbnailMouseOver.bind(null, card)}
    >
      <div className="small-movie-card__image">
        <img src={card.imgSrc} alt={card.imgDescription} width="280" height="175"/>
      </div>
      <h3 className="small-movie-card__title">
        <a
          className="small-movie-card__link"
          href={card.link}
          onClick={onThumbnailClick}
        >
          {card.title}
        </a>
      </h3>
    </article>
  );
};

MovieCardThumbnail.propTypes = {
  card: PropTypes.exact({
    id: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string
    ]),
    imgSrc: PropTypes.string,
    posterSrc: PropTypes.string,
    imgDescription: PropTypes.string,
    link: PropTypes.string,
    title: PropTypes.string,
    genre: PropTypes.string,
    year: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string
    ]),
    director: PropTypes.string,
    starring: PropTypes.arrayOf(PropTypes.string),
    rating: PropTypes.shape({
      score: PropTypes.string,
      level: PropTypes.string,
      count: PropTypes.number
    })
  }),
  onThumbnailClick: PropTypes.func.isRequired,
  onThumbnailMouseOver: PropTypes.func.isRequired
};

export {MovieCardThumbnail};
