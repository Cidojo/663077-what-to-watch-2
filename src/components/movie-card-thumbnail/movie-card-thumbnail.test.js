import * as React from 'react';
import {MovieCardThumbnail} from './movie-card-thumbnail.jsx';
import * as renderer from 'react-test-renderer';

it(`should render component without errors`, () => {
  const tree = renderer
    .create(<MovieCardThumbnail
      card={{
        id: 0,
        src: ``,
        imgSrc: ``,
        posterSrc: ``,
        imgDescription: ``,
        link: ``,
        title: ``,
        backgroundColor: ``,
        description: ``,
        isFavorite: false,
        previewSrc: ``,
        genre: ``,
        year: 0,
        director: ``,
        starring: [``],
        rating: 0,
        ratingCount: 0,
        runTime: 0
      }}

      onThumbnailClick={() => {}}
      onThumbnailMouseEnter={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
