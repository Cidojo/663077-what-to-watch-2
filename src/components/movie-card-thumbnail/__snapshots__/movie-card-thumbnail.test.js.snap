// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component without errors 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt=""
      height="175"
      src=""
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href=""
    >
      
    </a>
  </h3>
</article>
`;
